### PR TITLE
[Text Area] Text color and position of the text area counter

### PR DESCRIPTION
**Pull request by poetrainy**
This is the content of the conversation in [DDS-1419](https://daikinsvoil.atlassian.net/browse/DDS-1419?focusedCommentId=29396).

- Fixed the position of the text area counter within the Input Group.
- Fixed the text colour of the counter when an error occurs in the text area counter within the Input Group.

[DDS-1419]: https://daikinsvoil.atlassian.net/browse/DDS-1419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by poetrainy** [comment]
We will check with the designer about the error message that appears when the counter is exceeded.

File: src/components/input-group/daikin-input-group.ts [hunk 1]
```diff
@@ -206,6 +206,16 @@ export class DaikinInputGroup extends LitElement {
   @property({ type: Number, reflect: true, attribute: "textarea-max-count" })
   textareaMaxCount: number | null = null;
 
+  /**
+   * Maximum value to display on the counter. When `null`, the counter will be hidden.
```
Comment (yodas7) [inline src/components/input-group/daikin-input-group.ts:210]:
```suggestion
   * An error message displayed when the number of characters in the text area exceeds the limit (optional).
   * When the number of characters in the text area exceeds the limit, this is displayed below the label by overriding `helper` and `error`.
   * Only used when a text area is assigned to a slot and `textareaMaxCount` is set.
```

File: src/components/input-group/daikin-input-group.ts [hunk 2]
```diff
@@ -206,6 +206,16 @@ export class DaikinInputGroup extends LitElement {
...
+   */
+  @property({
+    type: String,
+    reflect: true,
+    attribute: "textarea-overflow-error",
+  })
+  textareaOverflowError: string | null = null;
```
Comment (yodas7) [inline src/components/input-group/daikin-input-group.ts:217]:
```suggestion
  textareaLimitExceedError: string | null = null;
```

File: src/components/input-group/daikin-input-group.ts [hunk 3]
```diff
@@ -206,6 +206,16 @@ export class DaikinInputGroup extends LitElement {
   @property({ type: Number, reflect: true, attribute: "textarea-max-count" })
   textareaMaxCount: number | null = null;
 
+  /**
+   * Maximum value to display on the counter. When `null`, the counter will be hidden.
+   */
+  @property({
+    type: String,
+    reflect: true,
+    attribute: "textarea-overflow-error",
```
Comment (yodas7) [inline src/components/input-group/daikin-input-group.ts:215]:
```suggestion
    attribute: "textarea-limit-exceed-error",
```

File: src/components/input-group/daikin-input-group.ts [hunk 4]
```diff
@@ -218,6 +228,14 @@ export class DaikinInputGroup extends LitElement {
   @state()
   private _textareaCount: number | null = null;
 
+  get textareaCounterOverflow(): boolean {
```
Comment (yodas7) [inline src/components/input-group/daikin-input-group.ts:231]:
```suggestion
  get textareaLimitExceeded(): boolean {
```

File: src/components/input-group/daikin-input-group.ts [hunk 5]
```diff
@@ -206,6 +206,16 @@ export class DaikinInputGroup extends LitElement {
   @property({ type: Number, reflect: true, attribute: "textarea-max-count" })
```
Comment (yodas7) [inline src/components/input-group/daikin-input-group.ts:206]: Update description.

```
The maximum number of characters that can be input into the text area.
If set, a counter will be displayed at the bottom of the text area.
Users can enter more characters than the limit, but in that case the counter will turn red and an error message will be displayed (if specified) to indicate that the limit has been exceeded.
```

File: src/components/input-group/daikin-input-group.ts [hunk 6]
```diff
@@ -260,26 +263,31 @@ export class DaikinInputGroup extends LitElement {
...
+      } else if (!!this.error && !this.disabled) {
+        return "error";
+      } else if (!!this.helper && this.disabled) {
+        return "helperDisabled";
+      } else if (this.helper) {
+        return "helper";
+      } else {
+        return "none";
+      }
+    };
```
Comment (yodas7) [inline src/components/input-group/daikin-input-group.ts:282]: Use IIFE.

```suggestion
    const helperTextVariant: HelperType = (() => {
      if (
        this.textareaLimitExceeded &&
        !!this.textareaLimitExceedError &&
        !this.disabled
      ) {
        return "textareaLimitExceedError";
      } else if (!!this.error && !this.disabled) {
        return "error";
      } else if (!!this.helper && this.disabled) {
        return "helperDisabled";
      } else if (this.helper) {
        return "helper";
      } else {
        return "none";
      }
    })();
```

**Review by ukkary — APPROVED** [review]

**Pull request merged**